tutorials/plugins/gdnative/gdnative-cpp-example.rst

```diff
@@ -0,0 +1,337 @@
+.. _doc_gdnative_cpp_example:
+
+GDNative C++ example
+====================
+
+Introduction
+------------
+This tutorial builds on top of the information given in the :ref:`GDNative C example <doc_gdnative_c_example>` so I highly recommend you read that first.
+
+The C++ bindings for GDNative are build on top of the nativescript GDNative API and provide a nicer way to "extend" nodes in Godot using C++. Basically this is the equivilant to writing GDScript scripts but in C++.
+
+We'll be looking at nativescript 1.0 which is available in Godot 3.0. Godot 3.1 will see the introduction of nativescript 1.1 which has a number of improvements. We'll update this tutorial once that is officially released but the overal structure is much the same.
+
+You can download the full example we'll be creating in this tutorial here: https://github.com/BastiaanOlij/gdnative_cpp_example
+
+Setting up your project
+-----------------------
+There are a few prerequisits you'll need:
```

Comment:
prerequisites

Also, link to http://docs.godotengine.org/en/latest/development/compiling/index.html here?